### PR TITLE
Azure v2 migration guide

### PR DESCRIPTION
**Pull request by mnlumi**
https://github.com/pulumi/pulumi-azure-native/issues/2497

### CODE REVIEW

File: themes/default/content/registry/packages/azure-native-v2/from-v1-to-v2.md [hunk 1]
```diff
@@ -0,0 +1,219 @@
+---
+title: Upgrade from Azure Native v1 to v2
+meta_desc: How to upgrade from v1 to v2 of the Pulumi Azure Native Provider.
```
Comment (susanev) [inline themes/default/content/registry/packages/azure-native-v2/from-v1-to-v2.md:3]:
```suggestion
meta_desc: How to upgrade from v1 to v2 of the Pulumi Azure Native Provider.
layout: package
```

**Review by susanev — COMMENTED** [review]
this will make the page have the left nav and table of contents on the right; instead of existing out in space

**Review by susanev — APPROVED** [review]

**Comment by susanev** [comment]
@mnlumi my api docs merge just gave you conflicts, ill go ahead and resolve them

File: themes/default/content/registry/packages/azure-native-v2/from-v1-to-v2/_index.md [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+- <PackageReference Include="Pulumi.AzureNative" Version="1.103.0"
++ <PackageReference Include="Pulumi.AzureNative" Version="2.0.0-beta.1"
+```
+
+{{% /choosable %}}
+{{% choosable language go %}}
+
+```go
+- github.com/pulumi/pulumi-azure-native-sdk/resources v1.103.0
++ github.com/pulumi/pulumi-azure-native-sdk/v2/storage v2.0.0-beta.1
```
Comment (mikhailshilkov) [inline themes/default/content/registry/packages/azure-native-v2/from-v1-to-v2/_index.md:64]: Change in module from `resources` to `storage`
```suggestion
+ github.com/pulumi/pulumi-azure-native-sdk/v2/resources v2.0.0-beta.1
```

File: themes/default/content/registry/packages/azure-native-v2/from-v1-to-v2/_index.md [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+```
+
+{{% /choosable %}}
+
+{{% /chooser %}}
+
+### Upgrade Imports
+
+Review your program for warnings on missing imports or deprecated resources. If your program contains any deprecated explicit versions, you will need to update these to a newer version. The suggested version is shown in the deprecation message.
```
Comment (mikhailshilkov) [inline themes/default/content/registry/packages/azure-native-v2/from-v1-to-v2/_index.md:73]: Shouldn't this happen while still on 1.103, so before upgrading to 2.0?

File: themes/default/content/registry/packages/azure-native-v2/from-v1-to-v2/_index.md [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+
+#### Pending Changes on Default Versions
+
+You may see pending changes when using the default version as the shape of the resource may have changed. You can choose to accept the changes, update your program to modify the resource properties to mitigate changes, or continue using the previous default version from v1.
+
+A full list of default version changes can be found in the [top-level resource versions](./top-level-resource-versions).
+
+To continue using the previous Azure API version of a resource:
+
+1. Check the documentation in your IDE or our [registry API docs](https://www.pulumi.com/registry/packages/azure-native-v2/) which identifies the previous version for each resource. For example: `Azure REST API Version: 2022-05-01. Prior API version in Azure Native 1.x: 2020-07-17-preview`
```
Comment (mikhailshilkov) [inline themes/default/content/registry/packages/azure-native-v2/from-v1-to-v2/_index.md:109]: Nit: should we align the versions in this example with the code snippets below?

File: themes/default/content/registry/packages/azure-native-v2/from-v1-to-v2/_index.md [hunk 5]
```diff
@@ -0,0 +1,218 @@
...
+});
+
+// Create a storage account that references the user-assigned managed identity
+const storageAccount = new storage.StorageAccount("mystorageaccount", {
+    resourceGroupName: resourceGroup.name,
+    kind: "StorageV2",
+    location: resourceGroup.location,
+    identity: {
+        type: "UserAssigned",
+        userAssignedIdentities: [userAssignedIdentity.id],
```
Comment (mikhailshilkov) [inline themes/default/content/registry/packages/azure-native-v2/from-v1-to-v2/_index.md:182]: Should we add examples in other languages?

File: themes/default/content/registry/packages/azure-native-v2/from-v1-to-v2/_index.md [hunk 6]
```diff
@@ -0,0 +1,218 @@
...
+
+In v2, the following resources are now associated with a `Flexible Server` variant instead of `Single Server` as they were in v1:
+- `Configuration`
+- `Database`
+- `FirewallRule`
+- `Server`
+- `PrivateEndpointConnection`
+
+Existing v1 programs upgrading to v2 using the default version will result in a change of resource type. For instance, `azure-native.dbformysql.Server` would previously have referred to a `Single Server` but will now refer to a `Flexible Server` and will result in a replacement of the resource during the next `pulumi up`. However, the properties of flexible servers are sufficiently different that, in a typed language, the program will not compile. If you would like to continue using the previous Azure API version, you may do so by using the previous explicit version.
```
Comment (mikhailshilkov) [inline themes/default/content/registry/packages/azure-native-v2/from-v1-to-v2/_index.md:214]: Can we name the explicit versions that have to be used, and maybe give a code example?

**Review by mikhailshilkov — COMMENTED** [review]
Looks great!

File: themes/default/content/registry/packages/azure-native-v2/from-v1-to-v2/_index.md (hunk 5 above)
Comment (mnlumi) [reply themes/default/content/registry/packages/azure-native-v2/from-v1-to-v2/_index.md:182]: I will add examples in other languages at a later date once [this](https://github.com/pulumi/pulumi/issues/13163) is resolved.

**Review by mikhailshilkov — APPROVED** [review]
:shipit:

**Pull request merged**